src/igvfd/types/file_set.py

```diff
@@ -441,29 +442,26 @@ def summary(self, request, file_set_type, input_file_sets=None, files=None, samp
-                            assay_term_name = assay_term_object.get('term_name', '')
-                            assay_titles.add(f"{assay_term_name} ({fileset_object['preferred_assay_title']})")
+                        target_assays = ['10x multiome', '10x multiome with MULTI-seq', 'SHARE-seq']
+                        preferred_assay_titles = fileset_object.get('preferred_assay_titles', [])
+                        assays_titles = fileset_object.get('assay_titles', [])
+                        if any(title in target_assays for title in preferred_assay_titles):
+                            assay_term_name.add(f"{', '.join(assays_titles)} ({', '.join(preferred_assay_titles)})")
                         else:
-                            assay_titles.add(fileset_object['preferred_assay_title'])
+                            assay_term_name.add(', '.join(preferred_assay_titles))
                     # Retrieve Measurement Sets associated with Auxiliary Sets.
                     elif input_fileset.startswith('/auxiliary-sets/'):
                         fileset_types.add(fileset_object['file_set_type'])
                         if 'measurement_sets' in fileset_object:
                             for candidate_fileset in fileset_object.get('measurement_sets'):
-                                measurement_set_object = request.embed(
-                                    candidate_fileset, '@@object?skip_calculated=true')
-                                assay_terms.add(measurement_set_object['assay_term'])
-                                if measurement_set_object['preferred_assay_title'] in ['10x multiome', '10x multiome with MULTI-seq', 'SHARE-seq']:
-                                    assay_term_object = request.embed(
-                                        measurement_set_object['assay_term'], '@@object?skip_calculated=true')
-                                    assay_term_name = assay_term_object.get('term_name', '')
-                                    assay_titles.add(
-                                        f"{assay_term_name} ({measurement_set_object['preferred_assay_title']})")
+                                target_assays = ['10x multiome', '10x multiome with MULTI-seq', 'SHARE-seq']
+                                preferred_assay_titles = fileset_object.get('preferred_assay_titles', [])
+                                assays_titles = fileset_object.get('assay_titles', [])
+                                if any(title in target_assays for title in preferred_assay_titles):
+                                    assay_term_name.add(
+                                        f"{', '.join(assays_titles)} ({', '.join(preferred_assay_titles)})")
                                 else:
-                                    assay_titles.add(measurement_set_object['preferred_assay_title'])
+                                    assay_term_name.add(', '.join(preferred_assay_titles))
```

Comment:
assay_term_names != preferred_assay_titles?

Reply:
np, i can change the variable name